movement/watch_faces/clock/french_revolutionary_face.c

```diff
@@ -0,0 +1,243 @@
+
+    char buf[11];
+    watch_date_time date_time;
+    fr_decimal_time decimal_time;
+
+    switch (event.event_type) {
+        case EVENT_ACTIVATE:
+            // Initial UI - Show a quick "splash screen"
+            watch_clear_display();
+            watch_display_string("FR  dECimL", 0);
+            break;
+        case EVENT_TICK:
+        case EVENT_LOW_ENERGY_UPDATE:
+            
+            date_time = watch_rtc_get_date_time();
+
+            decimal_time = get_decimal_time(&date_time);
+
+            set_display_buffer(buf, state, &decimal_time, &date_time);
+
+            // If we're in low-energy mode, don't write out the seconds. Also start the LE tick animation if it's not already going.
+            if (event.event_type == EVENT_LOW_ENERGY_UPDATE) {
+                buf[8] = ' ';
+                buf[9] = ' ';
+                if (!watch_tick_animation_is_running()) { watch_start_tick_animation(500); }
+            }
+
+            // Update the display with our decimal time
+            watch_display_string(buf, 0);
+
+            // Oh, and a one-off to set the colon after the "splash screen"
+            if (!state->colon_set_after_splash) {
+                watch_set_colon();
+                state->colon_set_after_splash = true;
+            }
+            break;
+        case EVENT_ALARM_BUTTON_UP:
+            state->display_type += 1 ; // cycle through the display types
+            if (state->display_type > 2) { state->display_type = 0; } // but return to 0 after 2
+            break;
+        case EVENT_ALARM_LONG_PRESS:
+            // I originally had chiming on the decimal-hour enabled, and this would enable/disable that chime, just like on
+            // the simple clock and decimal time faces. But because decimal seconds don't always line up with normal seconds,
+            // I assume the (decimal-)hourly chime could sometimes be missed. Additionally, I need this button for other purposes,
+            // now that I added seconds on/off toggle and upper normal-time with the ability to toggle that between 12/24hr format.
+            state->show_seconds = !state->show_seconds;
+            if (!state->show_seconds) { watch_display_string("  ", 8); }
+            else { watch_display_string("--", 8); }
+            break;
+        case EVENT_LIGHT_LONG_PRESS:
+            // In case anyone really wants that upper time in 12-hour format. I thought about using the global setting (settings->bit.clock_mode_24h)
+            // for this preference, but thought someone who prefers 12-hour format normally, might prefer 24hr when compared to a 10hr decimal day,
+            // so this is separate for now.
+            state->use_am_pm = !state->use_am_pm;
+            if (state->use_am_pm) {
+                watch_clear_indicator(WATCH_INDICATOR_24H);
+                if (date_time.unit.hour < 12) { watch_clear_indicator(WATCH_INDICATOR_PM); }
```

Comment:
`date_time` is being used uninitialized here, right? I don't think this is working the way it's intended